Userland/Demos/Mandelbrot/Mandelbrot.cpp

```diff
@@ -51,6 +51,20 @@ class MandelbrotSet {
         calculate();
     }
 
+    void pan_by(Gfx::IntPoint const& delta)
+    {
+        auto relative_width_pixel = (m_x_end - m_x_start) / m_bitmap->width();
+        auto relative_height_pixel = (m_y_end - m_y_start) / m_bitmap->height();
+
+        set_view(
+            m_x_start - delta.x() * relative_width_pixel,
+            m_x_end - delta.x() * relative_width_pixel,
+            m_y_start - delta.y() * relative_height_pixel,
+            m_y_end - delta.y() * relative_height_pixel);
+        correct_aspect();
```

Comment:
This shouldn't be necessary. You should only have to call `correct_aspect()` when changing the widget's size.